server/src/main/java/org/elasticsearch/repositories/MissingPluginRepository.java

```diff
@@ -0,0 +1,185 @@
+import org.elasticsearch.index.store.Store;
+import org.elasticsearch.indices.recovery.RecoveryState;
+import org.elasticsearch.snapshots.SnapshotId;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * This class represents a repository that still exists in the global cluster state but could not be loaded as its plugin was uninstalled.
+ */
+public class MissingPluginRepository extends AbstractLifecycleComponent implements Repository {
+
+    private final RepositoryMetadata repositoryMetadata;
+
+    public MissingPluginRepository(RepositoryMetadata repositoryMetadata) {
+        this.repositoryMetadata = repositoryMetadata;
+    }
+
+    private RepositoryPluginException createMissingPluginException() {
+        return new RepositoryPluginException(
+            repositoryMetadata.name(),
+            "the plugin is not installed for repository type [" + repositoryMetadata.type() + "]"
+        );
+    }
+
+    @Override
+    public RepositoryMetadata getMetadata() {
+        return repositoryMetadata;
+    }
+
+    @Override
+    public void getSnapshotInfo(GetSnapshotInfoContext context) {
+        throw createMissingPluginException();
+    }
+
+    @Override
+    public Metadata getSnapshotGlobalMetadata(SnapshotId snapshotId) {
+        throw createMissingPluginException();
+    }
+
+    @Override
+    public IndexMetadata getSnapshotIndexMetaData(RepositoryData repositoryData, SnapshotId snapshotId, IndexId index) throws IOException {
+        throw createMissingPluginException();
+    }
+
+    @Override
+    public void getRepositoryData(ActionListener<RepositoryData> listener) {
+        listener.onFailure(createMissingPluginException());
+    }
+
+    @Override
+    public void finalizeSnapshot(FinalizeSnapshotContext finalizeSnapshotContext) {
+        finalizeSnapshotContext.onFailure(createMissingPluginException());
+    }
+
+    @Override
+    public void deleteSnapshots(
+        Collection<SnapshotId> snapshotIds,
+        long repositoryStateId,
+        Version repositoryMetaVersion,
+        ActionListener<RepositoryData> listener
+    ) {
+        listener.onFailure(createMissingPluginException());
+    }
+
+    @Override
+    public long getSnapshotThrottleTimeInNanos() {
+        throw createMissingPluginException();
+    }
+
+    @Override
+    public long getRestoreThrottleTimeInNanos() {
+        throw createMissingPluginException();
+    }
+
+    @Override
+    public String startVerification() {
+        throw createMissingPluginException();
+    }
+
+    @Override
+    public void endVerification(String verificationToken) {
+        throw createMissingPluginException();
+    }
+
+    @Override
+    public void verify(String verificationToken, DiscoveryNode localNode) {
+        throw createMissingPluginException();
+    }
+
+    @Override
+    public boolean isReadOnly() {
+        return true;
```

Comment:
Currently this result in:
```
PUT http://localhost:9200/_snapshot/repository-hdfs/test

>>>
{
  "error": {
    "root_cause": [
      {
        "type": "repository_exception",
        "reason": "[repository-hdfs] cannot create snapshot in a readonly repository"
      }
    ],
    "type": "repository_exception",
    "reason": "[repository-hdfs] cannot create snapshot in a readonly repository"
  },
  "status": 500
}
```

I do not think this is correct and looking for a way to return no plugin exception when taking a snapshot.

Reply:
Resolved: 

```
http://localhost:9200/_snapshot/repository-hdfs/test

>>>
HTTP/1.1 500 Internal Server Error
X-elastic-product: Elasticsearch
content-type: application/json;charset=utf-8

{
  "error": {
    "root_cause": [
      {
        "type": "repository_plugin_exception",
        "reason": "[repository-hdfs] the plugin is not installed for repository type [hdfs]"
      }
    ],
    "type": "repository_plugin_exception",
    "reason": "[repository-hdfs] the plugin is not installed for repository type [hdfs]"
  },
  "status": 500
}
```